lib/postgrex/utils.ex

```diff
@@ -63,9 +63,16 @@ defmodule Postgrex.Utils do
     [version_string | _] = String.split(version, " ", parts: 2)
     segments =
       version_string
-      |> String.trim_trailing("devel")
-      |> String.split(".", parts: 4)
-      |> Enum.map(&String.to_integer/1)
+      |> String.split(" ")
+      |> hd()
+      |> String.split(~r/\.|(?<=\d)(?=[^\d.])|(?<=[^\d.])(?=\d)/, parts: 4)
```

Comment:
FWIW, this regex matches:

- dots
- boundaries between digits and neither digits nor dots
- boundaries between neither digits nor dots and digits

It preserves the existing behaviour but also splits on boundaries.

Reply:
Thanks! I would prefer to not have to rely on a regex though. It will be quite hard to maintain and update it in the future. Maybe we can achieve this with "regular" code?

Reply:
You brought `Regex` to _Elixir_, not me :)

Pick one, please:

```elixir
  defp do_parse(s, acc) do
    case Integer.parse(s) do
      {i, ""} ->
        [i | acc]

      {i, " " <> _} ->
        [i | acc]

      {i, "." <> rest} ->
        do_parse(rest, [i | acc])

      {i, <<c::binary-size(1), rest::binary()>>} ->
        [ver] = String.to_charlist(c)
        do_parse(rest, [ver - 255, i | acc])

      :error ->
        with <<_::binary-size(1), rest::binary()>> <- s, do: do_parse(rest, acc)
    end
  end
```
or
```elixir
  def do_parse(s) do
    s
    |> String.codepoints()
    |> Enum.reduce_while({false, []}, fn
      d, {false, acc} when d in ~w|0 1 2 3 4 5 6 7 8 9 0| ->
        {:cont, {true, [d | acc]}}

      d, {true, [h | acc]} when d in ~w|0 1 2 3 4 5 6 7 8 9 0| ->
        {:cont, {true, [h <> d | acc]}}

      ".", {true, acc} ->
        {:cont, {false, acc}}

      " ", acc ->
        {:halt, acc}

      any, {true, acc} ->
        {:cont,
         {false, [any |> String.to_charlist() |> hd() |> Kernel.-(255) |> to_string() | acc]}}

      _, acc ->
        {:cont, acc}
    end)
    |> elem(1)
    |> Enum.map(&String.to_integer/1)
  end
```
called as 
```elixir
    segments =
      version_string
      |> do_parse()
      |> Enum.reverse()
```

Am I missing some obvious easy solution?